packages/clerk-js/src/ui/elements/FormControl.tsx

```diff
@@ -53,6 +64,57 @@ const getInputElementForType = (type: FormControlProps['type']) => {
   return CustomInputs[customInput] || Input;
 };
 
+function useFormTextAnimation() {
+  const prefersReducedMotion = usePrefersReducedMotion();
+
+  const getFormTextAnimation = useCallback(
+    (enterAnimation: boolean): ThemableCssProp => {
+      if (prefersReducedMotion) {
+        return {
+          animation: 'none',
+        };
+      }
+      return t => ({
+        animation: `${enterAnimation ? animations.inAnimation : animations.outAnimation} ${
+          t.transitionDuration.$slower
+        } ${t.transitionTiming.$common}`,
+        transition: `height ${t.transitionDuration.$slow}  ease`, // This is expensive but required for a smooth layout shift
+      });
+    },
+    [prefersReducedMotion],
+  );
+
+  return {
+    getFormTextAnimation,
+  };
+}
+
+type Px = number;
+const useCalculateErrorTextHeight = () => {
+  const [height, setHeight] = useState<Px>(24);
+
+  const calculateHeight = useCallback((element: HTMLElement | null, messageToDisplay: string | undefined) => {
+    if (element) {
+      const fontSize = parseInt(getComputedStyle(element).fontSize.replace('px', ''));
```

Comment:
⛔ Please invoke `getComputedStyle` once for the same element, as it [causes layout thrashing](https://gist.github.com/paulirish/5d52fb081b3570c81e3a#calling-getcomputedstyle).

Reply:
I'd also expect us using requestAnimationFrame.

Reply:
We will get an new reference when an element mounts, unmounts and mounts again. 
So I'd propose to simply use a useRef and store the font size there for all elements. This will ensure the getComputedStyle is called once. 
Obviously this works under the assumption that all elements will share the same styles which in our case it does.